frontend/src/Components/Link/Link.tsx

```diff
@@ -1,96 +1,87 @@
 import classNames from 'classnames';
 import React, {
-  ComponentClass,
-  FunctionComponent,
+  ComponentPropsWithoutRef,
+  ElementType,
   SyntheticEvent,
   useCallback,
 } from 'react';
 import { Link as RouterLink } from 'react-router-dom';
 import styles from './Link.css';
 
-interface ReactRouterLinkProps {
-  to?: string;
-}
+export type LinkProps<C extends ElementType = 'button'> =
```

Comment:
Couldn't do it with an `interface`

![image](https://github.com/user-attachments/assets/ea53d468-c7b5-461e-bbc5-cbe258286277)